minio/api.py

```diff
@@ -1695,13 +1706,20 @@ def _get_bucket_region(self, bucket_name):
         :param bucket_name: Bucket name for which region will be fetched.
         :return: Region of bucket name.
         """
+
+        # Region set in constructor, return right here.
+        if self._region:
+            return self._region
```

Comment:
region is provided to override all the buckets and all calls.. There is no cache based approach if region is set.